plugins/inventory/icinga2.py

```diff
@@ -177,24 +202,35 @@ def _convert_inv(self, json_data):
         """Convert Icinga2 API data to JSON format for Ansible"""
         groups_dict = {"_meta": {"hostvars": {}}}
         for entry in json_data:
-            host_name = entry['name']
             host_attrs = entry['attrs']
+            if self.inventory_attr == "name":
+                host_name = entry.get('name')
+            if self.inventory_attr == "address":
+                # When looking for address for inventory, if missing fallback to object name
+                if host_attrs.get('address') != '':
```

Comment:
Otherwise you can end up with `None`:
```suggestion
                if host_attrs.get('address', '') != '':
```

Reply:
Fixed, thank you,